### ISSUE TITLE
technical question

### ISSUE THREAD

**Issue by SebastianJanampa** [roboflow/rf-detr#24]
Hello,

Thanks for such incredible work. Combining DINOv2 with DETR is a mind-blowing idea.
Did you make any modifications apart from the backbone?
Also, in terms of comparison, will you run experiments with RT-DETR on the ROBOFLOW dataset?
You mentioned you had problems fine-tuning D-FINE, did you also face the same problem with [DEIM-DFINE](https://github.com/ShihuaHuang95/DEIM)?

Regards,
Sebastian

**Comment by probicheaux** [comment]
Architecture changes:
1. Added windows to the Dinov2
2. Replaced all batch norms in projector with layer norms so grad accum will work so users can train on smaller gpus

We haven't ran deim-dfine or rt-detr on RF100-VL but we definitely should

**Comment by isaacrob-roboflow** [comment]
Those are the major changes. The rest of the changes will be made clear when we release our preprint!

**Comment by SebastianJanampa** [comment]
Thank you so much @probicheaux  and @isaacrob-roboflow, for your responses. When will the preprint be released?

**Comment by anyaflya** [comment]
Hello, interesting to see what are the main changes in the architecture! 
I am also curious about the preprint, in order to understand better the details of RF-DETR architecture. Could you update us about it, @isaacrob-roboflow and @probicheaux ?

**Comment by isaacrob-roboflow** [comment]
Preprint coming soon!

Also @SebastianJanampa we got D-FINE training successfully thanks to @Matvezy

**Comment by SebastianJanampa** [comment]
@isaacrob-roboflow , that's awesome.

Could you provide more information about how you trained D-FINE?

**Comment by Matvezy** [comment]
Hi @SebastianJanampa we used the o365 checkpoints for S, M, and L. And the coco checkpoint for N as o365 is not available. 

As discussed before we had a problem of getting mAP of 0 for every dataset with fine-tuning D-FINE using their config here: https://github.com/Peterande/D-FINE/tree/master/configs/dfine/custom/objects365

However we discovered that using their basic custom config solved it and produced sensible results: https://github.com/Peterande/D-FINE/tree/master/configs/dfine/custom

We standardized all of the models that we benchmarked to train for 100 epochs and use the batch size 16. So we modified the configs to have epochs: 100, changed the related values like stop_epoch to 90 proportionally (rounded to 90 for simplicity), and set the batch size to 16 in all configs. We also experimented with halving the lr to account for twofold batch size reduction, but it lead to worse results so we left it out. We left everything else the same.

I think the main difference between the o365 and the default custom configs are the following fields:

`pretrained: False

  collate_fn:
    stop_epoch: 48 (90 with our mdofications)
    ema_restart_decay: 0.9999
    base_size_repeat: 6

ema:
  warmups: 0

lr_warmup_scheduler:
  warmup_duration: 0`

We didn't have enough resources/time to ablate the individual effects of those and just went with the default config. If that's something you would like to benchmark on your end we would love to see the results! 
Hope this helps, let me know if you'll have any questions :)

**Comment by SebastianJanampa** [comment]
Hi @Matvezy,

Thank you for such a detailed answer. It is very interesting to find out that D-FINE training is very sensitive. 

Would it also be possible to provide the commands to replicate RF-DETR models to reproduce COCO results? How many epochs would i need to set?

**Comment by isaacrob-roboflow** [comment]
Preprint is now live here :) https://arxiv.org/abs/2511.09554
